internal/internal_workflow_client.go

```diff
@@ -260,6 +260,40 @@ func (wc *workflowClient) ExecuteWorkflow(ctx context.Context, options StartWork
 	}, nil
 }
 
+// GetWorkflow gets a workflow execution and returns a WorkflowRun that will allow you to wait until this workflow
+// reaches the end state, such as workflow finished successfully or timeout.
+// The current timeout resolution implementation is in seconds and uses math.Ceil(d.Seconds()) as the duration. But is
+// subjected to change in the future.
+func (wc *workflowClient) GetWorkflow(ctx context.Context, workflowID string, runID string) (WorkflowRun, error) {
+
+	// retrieve the workflow execution information
+	resp, err := wc.DescribeWorkflowExecution(ctx, workflowID, runID)
+	if err != nil {
+		return nil, err
+	}
+
+	// get the current run ID
+	if resp.WorkflowExecutionInfo != nil &&
+		resp.WorkflowExecutionInfo.Execution != nil {
+		workflowID = resp.WorkflowExecutionInfo.Execution.GetWorkflowId()
+		runID = resp.WorkflowExecutionInfo.Execution.GetRunId()
+	} else {
+		return nil, errors.New("describeWorkflowExecution did not return workflowID and runID")
```

Comment:
Is it even possible?

Reply:
From the current implementation of server no, but just from the RPC perspective yes.. I can drop that check and assume response always exists..